fossunited/fossunited/forms.py

```diff
@@ -7,14 +7,32 @@
 """
 
 
+def is_valid_doctype(doctype: str):
+    """
+    Validate the doctype to be either "FOSS Event CFP Submission" or "FOSS Event RSVP Submission"
+    """
+    if doctype not in [
+        "FOSS Event RSVP Submission",
+        "FOSS Event CFP Submission",
+    ]:
+        return False
+
+    return True
+
+
 @frappe.whitelist()
-def create_submission(fields):
+def create_submission(doctype: str, fields: dict):
     """
     Used for RSVP and CFPS
     Create a new submission for the given doctype with the given fields.
     Fields variable should have the doctype values as well
     """
-    doc = frappe.get_doc(fields)
+    if not is_valid_doctype(doctype):
+        frappe.log("Unauthorized usage of create submission")
+        frappe.msgprint("Not allowed")
+        return
+
+    doc = frappe.get_doc(doctype, fields)
     doc.insert(ignore_permissions=True)
```

Comment:
Why are we taking the DocType at all as argument here? Let's have separate endpoints if absolutely needed.